ceno_zkvm/benches/fibonacci.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    fs,
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+
+use ceno_emul::{CENO_PLATFORM, Platform, Program, WORD_SIZE};
+use ceno_zkvm::{
+    self,
+    e2e::{run_e2e_gen_witness, run_e2e_proof},
+};
+use criterion::*;
+
+use goldilocks::GoldilocksExt2;
+use mpcs::BasefoldDefault;
+
+criterion_group! {
+  name = op_add;
+  config = Criterion::default().warm_up_time(Duration::from_millis(20000));
+  targets = bench_e2e
+}
+
+criterion_main!(op_add);
+
+const NUM_SAMPLES: usize = 10;
+
+fn bench_e2e(c: &mut Criterion) {
+    type Pcs = BasefoldDefault<E>;
+    let mut file_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    file_path.push("examples/fibonacci.elf");
+    let stack_size = 32768;
+    let heap_size = 2097152;
+    let elf_bytes = fs::read(&file_path).expect("read elf file");
+    let program = Program::load_elf(&elf_bytes, u32::MAX).unwrap();
+
+    // use sp1 platform
+    let platform = Platform {
+        // The stack section is not mentioned in ELF headers, so we repeat the constant STACK_TOP here.
+        stack_top: 0x0020_0400,
+        rom: program.base_address
+            ..program.base_address + (program.instructions.len() * WORD_SIZE) as u32,
+        ram: 0x0010_0000..0xFFFF_0000,
+        unsafe_ecall_nop: true,
+        ..CENO_PLATFORM
+    };
+
+    for max_steps in [1usize << 20, 1usize << 21, 1usize << 22] {
```

Comment:
Do you think it makes sense to have a smaller number like `1 << 10` as well? It works much faster and it can be a good sanity check. 

By the way, do you know a simple way to not run for all three sizes every time? Or even to pass size from the command-line?

Reply:
For "1 << 20" per iteration it take ~ 20 secs to warm up (on ceno server), so for sanity check purpose it's also fine without wait too long time. (most of the time are on key step actually, 1 << 20 proof time is < 5 secs)

For run with custom start/end, I tried to have Args + pass from CLI, but unfortunately the argument parsing will conflicts with `Criteria` crate, both complain unexpected argument, atm I dont have idea, will try look back to it later :)